pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go

```diff
@@ -0,0 +1,162 @@
+		return c.remove(ctx)
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	// Confirm that the ManagedProxyServiceResolver API is installed before proceding
+	_, err = c.crdClient.Get(ctx, api.ManagedProxyServiceResolverCRDName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		klog.V(4).Info("Skipping thanos-querier ManagedProxyServiceResolver sync. API is missing./n")
+		return nil
+	}
+
+	if err != nil {
+		klog.V(4).Infof("Error getting ManagedProxyServiceResolver CRD: %s/n", err)
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("ThanosQuerierProxyServiceResolverSync", "GetManagedProxyServiceResolverCRDError", err))
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	reason, err := c.SyncThanosQuerierServiceResolver(ctx)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("ThanosQuerierProxyServiceResolverSync", reason, err))
+	return statusHandler.FlushAndReturn(err)
+}
+
+func (c *ThanosQuerierProxyResolverController) SyncThanosQuerierServiceResolver(ctx context.Context) (string, error) {
+	required := managedproxyserviceresolver.DefaultThanosQuerierProxyServiceResolver()
+	existing, err := c.managedProxyServiceResolverClient.Get(ctx, api.ThanosQuerierProxyServiceResolverName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err = c.managedProxyServiceResolverClient.Create(ctx, required, metav1.CreateOptions{})
+		if err != nil {
+			klog.V(4).Infof("error creating initial thanos-querier ManagedServiceProxyResolver: %s/n", err)
+			return "CreateManagedProxyServiceResolverError", err
+		}
+		return "", nil
+	}
+
+	if err != nil || existing == nil {
+		klog.V(4).Infof("error getting existing thanos-querier ManagedServiceProxyResolver: %s/n", err)
+		return "GetManagedProxyServiceResolverError", err
+	}
+
+	// Check that existing matches required, delete and recreate if not
+	existingCopy := existing.DeepCopy()
+	specEqual := equality.Semantic.DeepEqual(existingCopy.Spec, required.Spec)
+	modified := resourcemerge.BoolPtr(false)
+	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
+
+	// Delete and recreate if it has been modified
+	if *modified || !specEqual {
+		err = c.managedProxyServiceResolverClient.Delete(ctx, api.ThanosQuerierProxyServiceResolverName, metav1.DeleteOptions{})
+		if err != nil {
+			klog.V(4).Infof("error deleting existing thanos-querier ManagedServiceProxyResolver: %s/n", err)
+			return "UpdateManagedProxyServiceResolverError", err
+		}
+		_, err = c.managedProxyServiceResolverClient.Create(ctx, required, metav1.CreateOptions{})
+		if err != nil {
+			klog.V(4).Infof("error creating updated thanos-querier ManagedServiceProxyResolver: %s/n", err)
+			return "UpdateManagedProxyServiceResolverError", err
+		}
+	}
+	return "", nil
+}
+
+func (c *ThanosQuerierProxyResolverController) remove(ctx context.Context) error {
```

Comment:
nit. just to stay consistant with the other remove methods that our controllers have
```suggestion
func (c *ThanosQuerierProxyResolverController) removeManagedProxyServiceResolver(ctx context.Context) error {
```